engine/access/rpc/backend/backend.go

```diff
@@ -226,14 +225,51 @@ func executionNodesForBlockID(
 	state protocol.State) (flow.IdentityList, error) {
 
 	// lookup the receipts storage with the block ID
-	receipts, err := executionReceipts.ByBlockIDAllExecutionReceipts(blockID)
+	allReceipts, err := executionReceipts.ByBlockIDAllExecutionReceipts(blockID)
 	if err != nil {
 		return nil, fmt.Errorf("failed to retreive execution receipts for block ID %v: %w", blockID, err)
 	}
 
+	// execution result ID to execution receipt map to keep track of receipts by their result id
+	var identicalReceipts = make(map[flow.Identifier][]*flow.ExecutionReceipt)
+
+	// maximum number of matching receipts found so far for any execution result id
+	maxMatchedReceiptCnt := 0
+	// execution result id key for the highest number of matching receipts in the identicalReceipts map
+	var maxMatchedReceiptResultID flow.Identifier
+
+	// find the largest list of receipts which have the same result ID
+	for _, receipt := range allReceipts {
+
+		resultID := receipt.ExecutionResult.ID()
+		receipts, found := identicalReceipts[resultID]
+
+		if !found {
+			// first of its kind
+			receipts = []*flow.ExecutionReceipt{receipt}
+			identicalReceipts[resultID] = receipts
+			// keep looking for at least one more receipt with the same result
+			// since we need at least two execution receipt which have the same result
+			continue
+		}
+
+		// append receipt to the list of all receipts for this resultID
+		receipts = append(receipts, receipt)
+		identicalReceipts[resultID] = receipts
+
+		currentMatchedReceiptCnt := len(receipts)
+		if currentMatchedReceiptCnt > maxMatchedReceiptCnt {
+			maxMatchedReceiptCnt = currentMatchedReceiptCnt
+			maxMatchedReceiptResultID = resultID
+		}
+	}
+
+	// pick the largest list of matching receipts
+	matchingReceipts := identicalReceipts[maxMatchedReceiptResultID]
```

Comment:
We need to pick the largest that is >=2, not just the largest. Otherwise if we receive the faulty receipt first and call this method, the largest will have only the faulty receipt in it.

```suggestion
	// pick the largest list of matching receipts
	matchingReceipts := identicalReceipts[maxMatchedReceiptResultID]
	if len(matchingReceipts) < 2 {
		return nil, someerror
	}
```

⚠️ Would actually be better to do this further down in https://github.com/onflow/flow-go/pull/428/files#diff-ac13d40a77a065ec65a685ed828d76f849a988759b47ce8c2b6f1a8750b7aed6R287. That way we have extra certainty the executor IDs are distinct, not just the receipts.

Reply:
the `continue` on line 253 would avoid executing this code if this was the very first receipt.

Reply:
That continue isn't there any more

Reply:
But handled on line 286 👌